MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,20 +26,56 @@ import Foundation
 
 extension MSALNativeAuthUserAccountResult {
 
-    func getAccessTokenInternal(
-        forceRefresh: Bool,
-        correlationId: UUID?,
-        cacheAccessor: MSALNativeAuthCacheInterface
-    ) async -> MSALNativeAuthCredentialsControlling.RefreshTokenCredentialControllerResponse {
+    func getAccessTokenInternal(forceRefresh: Bool = false,
+                                scopes: [String],
+                                correlationId: UUID? = nil,
+                                delegate: CredentialsDelegate) {
+
+        let params = MSALSilentTokenParameters(scopes: scopes, account: account)
         let context = MSALNativeAuthRequestContext(correlationId: correlationId)
-        let correlationId = context.correlationId()
-
-        if forceRefresh || self.authTokens.accessToken.isExpired() {
-            let controllerFactory = MSALNativeAuthControllerFactory(config: configuration)
-            let credentialsController = controllerFactory.makeCredentialsController(cacheAccessor: cacheAccessor)
-            return await credentialsController.refreshToken(context: context, authTokens: authTokens, userAccountResult: self)
-        } else {
-            return .init(.success(MSALNativeAuthTokenResult(authTokens: authTokens)), correlationId: correlationId)
+        params.forceRefresh = forceRefresh
+        params.correlationId = correlationId
+
+        let challengeTypes = MSALNativeAuthPublicClientApplication.convertChallengeTypes(configuration.challengeTypes)
+        let authority = try? MSALCIAMAuthority(url: configuration.authority.url)
+        let config = MSALPublicClientApplicationConfig(clientId: configuration.clientId,
+                                                       redirectUri: configuration.redirectUri,
+                                                       authority: authority)
+
+        guard let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
+            MSALLogger.log(
+                            level: .error,
+                            context: context,
+                            format: "Config or challenge types unexpectedly found nil."
+                        )
+            Task { await delegate.onAccessTokenRetrieveError(
+                error: RetrieveAccessTokenError(type: .generalError,
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { result, error in
+
+            if let error = error as? NSError {
+                let accessTokenError = RetrieveAccessTokenError(type: .generalError,
+                                                                correlationId: result?.correlationId ?? UUID(),
```

Comment:
Can you use the context here?
```suggestion
                                                                correlationId: result?.correlationId ?? context.correlationId(),
```

Reply:
Sure, done.